### PR TITLE
feat: add camp overview footer (camp deletion)

### PR DESCRIPTION
**Pull request by williamtran10**
## Notion ticket link

[Camp Overview: Footer + Actions](https://www.notion.so/uwblueprintexecs/Camp-Overview-Footer-Actions-9ef2fa77aa924218b4a425ba60d37b43)

## Implementation description
* Added delete action to camp overview footer

## Steps to test
1. Create new camp
2. Go to camp overview page for said camp and use the delete flow

## What should reviewers focus on?
* I used a state to conditionally render the Redirect - let me know if there's a better way to do that.

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/CampOverview/FooterDeleteModal.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      <ModalOverlay />
+      <ModalContent>
+        {redirect && (
+          <Redirect
+            to={{
+              pathname: CAMPS_PAGE,
+            }}
+          />
+        )}
+        <ModalHeader>{`Delete ${camp.name}`}</ModalHeader>
```
Comment (e-wai) [inline frontend/src/components/pages/CampOverview/FooterDeleteModal.tsx:66]: nit: Does this work? 

```ts
<ModalHeader>Delete {camp.name}</ModalHeader> 
```

File: frontend/src/APIClients/CampsAPIClient.ts [hunk 2]
```diff
@@ -36,7 +36,19 @@ const editCampById = async (
   }
 };
 
+const deleteCamp = async (id: string): Promise<boolean> => {
+  try {
+    await baseAPIClient.delete(`/camp/${id}`, {
+      headers: { Authorization: BEARER_TOKEN },
```
Comment (e-wai) [inline frontend/src/APIClients/CampsAPIClient.ts:42]: For cleanliness - can we create method in the `CampsAPIClient` for this?

**Review by e-wai — APPROVED** [review]
Some comments, nothing blocking

File: frontend/src/components/pages/CampOverview/FooterDeleteModal.tsx (hunk 1 above)
Comment (williamtran10) [reply frontend/src/components/pages/CampOverview/FooterDeleteModal.tsx]: Done now

File: frontend/src/APIClients/CampsAPIClient.ts (hunk 2 above)
Comment (williamtran10) [reply frontend/src/APIClients/CampsAPIClient.ts:42]: What do you mean by this?

Comment (e-wai) [reply frontend/src/APIClients/CampsAPIClient.ts:42]: Oh never mind, I don't know what I thought I saw ❓ 

This looks good

**Pull request merged**